lib.rs

```diff
@@ -322,8 +414,7 @@ impl ThreadPool {
         }
     }
 
-    /// Block the current thread until all jobs in the pool are completed.
-    /// Many threads can wait for a pool to finish concurrently.
+    /// Block the current thread until all jobs in the pool have been executed.
```

Comment:
I would add a text like this: 


A pool can be joined repeatedly. 

Calling the join method on an empty pool will cause it to return immediately.

The join method may be used from multiple threads at the same time.

# Warning

Joining the pool from within the pool will cause a deadlock.
This behavior is considered safe.